test/extended/dns/dns.go

```diff
@@ -130,21 +148,13 @@ func validateDNSResults(f *e2e.Framework, pod *api.Pod, fileNames sets.String, e
 		e2e.Failf("Failed to create %s pod: %v", pod.Name, err)
 	}
 
-	Expect(f.WaitForPodRunning(pod.Name)).To(BeNil())
-	Expect(wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
-		pod, err := podClient.Get(pod.Name)
-		if err != nil {
-			return false, err
-		}
-		switch pod.Status.Phase {
-		case api.PodSucceeded:
-			return true, nil
-		case api.PodFailed:
-			return false, fmt.Errorf("pod failed")
-		default:
-			return false, nil
-		}
-	})).To(BeNil())
+	w, err := f.Client.Pods(f.Namespace.Name).Watch(api.SingleObject(api.ObjectMeta{Name: pod.Name}))
```

Comment:
Yeah

On Mon, Jul 11, 2016 at 12:57 PM, Maciej Szulik <notifications@github.com>
wrote:

> In test/extended/dns/dns.go
> <https://github.com/openshift/origin/pull/9777#discussion_r70295415>:
>
> > -	Expect(f.WaitForPodRunning(pod.Name)).To(BeNil())
> > -	Expect(wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
> > -		pod, err := podClient.Get(pod.Name)
> > -		if err != nil {
> > -			return false, err
> > -		}
> > -		switch pod.Status.Phase {
> > -		case api.PodSucceeded:
> > -			return true, nil
> > -		case api.PodFailed:
> > -			return false, fmt.Errorf("pod failed")
> > -		default:
> > -			return false, nil
> > -		}
> > -	})).To(BeNil())
> > +	w, err := f.Client.Pods(f.Namespace.Name).Watch(api.SingleObject(api.ObjectMeta{Name: pod.Name}))
>
> This is the problem upstream, the Watch doesn't have ResourveVersion set
> which causes us missing certain events, can you add it?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9777/files/e4177ba24d1a43882cafa5c390380da2cf0a7ef2#r70295415>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_pyEH8QSXVfRQPW4lJRK9PGfR0pqDks5qUnXggaJpZM4JJhZ7>
> .
>